GitCommands/Git/GitModule.cs

```diff
@@ -2470,10 +2469,49 @@ public string GetEffectiveSetting(string setting)
             return EffectiveConfigFile.GetValue(setting);
         }
 
-        public string GetEffectiveGitSetting(string setting, bool cache = true)
+        public GitConfigGetResult GetEffectiveGitSetting(string setting, bool cache = true)
         {
-            GitArgumentBuilder args = new("config") { "--includes", "--get", setting };
-            return GitExecutable.GetOutput(args, cache: cache ? GitCommandCache : null).Trim();
+            GitArgumentBuilder args = new("config")
+            {
+                "--includes",
+                "--get",
+                "-z",
+                "--show-scope",
+                "--show-origin",
+                setting
+            };
+            ExecutionResult result = GitExecutable.Execute(args, cache: cache ? GitCommandCache : null, throwOnErrorExit: false); // Result codes are handled below
+
+            if (!result.ExitCode.HasValue)
+            {
+                // unlikely , but possible and allows for use of ExitCode.Value since we know HasValue is true
+                throw new ExternalOperationException("git config", args.ToString(), WorkingDir, result.ExitCode, new InvalidOperationException("Exit code had no value"));
+            }
+
+            string value = result.StandardOutput?.Trim() ?? "" + Delimiters.Null;
+
+            // Scope, file, value split by null.
+            string[] values = value.Split(Delimiters.Null, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I'll fix. Realized I shouldn't remove empty here. Just a by default thing I do because it makes sense most times.

Reply:
Can we use spans here to reduce allocations?

Reply:
Sure.

Reply:
![image](https://github.com/gitextensions/gitextensions/assets/3629489/f380f565-5f65-4e99-b13c-45767a18f4f3)
![image](https://github.com/gitextensions/gitextensions/assets/3629489/54af8e83-5e06-460d-a781-c67d9cc78d7d)